### PR TITLE
[VACMS-19449]Situation updates static banner

### PR DESCRIPTION
**Pull request by SnowboardTechie**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Adds a situation updates banner component that when enabled, takes the div prepared by content-build and adds a banner for situation updates. Currently the banner is working with static data and disabled behind a feature flag. Future work will enable this banner on only the appropriate pages (locations with situation updates fetched from vets-api).
- Sitewide team
- This is kept behind the flipper flag `banner_use_alternative_banners`
- Includes update to watch to setup with 12GB as 8 was not enough to complete builds

## Related issue(s)

- Issue: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19449
- Epic: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19302

## Testing done

- This introduces no noticeable changes to users as long as the flipper remains disabled
- When flipper is enabled, it updates the empty situation updates banner with the static data in the property config

## Scre
Screenshot 2024-11-07 at 10 21 02 AM
enshots

## What areas of the site does it impact?
This is interacting with a header level feature so it has the potential for sitewide impact

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by eselkin — DISMISSED** [review]
Nice use of the feature toggles in the tests.

**Comment by eselkin** [comment]
Also, thanks for putting in the memory updates for watch. I thought I did, but seemed to have missed it somewhere in my changes.

File: src/applications/static-pages/situation-updates-banner/createSituationUpdatesBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const alternativeBannersEnabled = useToggleValue(
+    TOGGLE_NAMES.bannerUseAlternativeBanners,
+  );
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+
+  if (isLoadingFeatureFlags || !alternativeBannersEnabled) {
+    return null;
+  }
+
+  const defaultProps = {
```
Comment (randimays) [inline src/applications/static-pages/situation-updates-banner/createSituationUpdatesBanner.jsx:24]: The banner component uses these props: `id, alertType, headline, showClose, content`. Where are all the other props in this object used?

Comment (SnowboardTechie) [reply src/applications/static-pages/situation-updates-banner/createSituationUpdatesBanner.jsx:24]: These end up being used by the Banner component itself `src/applications/static-pages/situation-updates-banner/situationUpdateBanner.jsx`

Comment (SnowboardTechie) [reply src/applications/static-pages/situation-updates-banner/createSituationUpdatesBanner.jsx:24]: They aren't being used in this static V1 version but are all pieces that are intended to be fetched and stored in the DB. I expect we may want to remove a few of these before this is 100% done, but I'm open to removing the extras for now if that feels too crufty

Comment (randimays) [reply src/applications/static-pages/situation-updates-banner/createSituationUpdatesBanner.jsx:24]: Thanks for clearing this up! All good.

**Review by randimays — DISMISSED** [review]
LGTM!

File: src/applications/static-pages/static-pages-entry.js [hunk 2]
```diff
@@ -200,6 +201,7 @@ createScheduleViewVAAppointmentsPage(
   widgetTypes.SCHEDULE_VIEW_VA_APPOINTMENTS_PAGE,
 );
 createSecureMessagingPage(store, widgetTypes.SECURE_MESSAGING_PAGE);
+createSituationUpdatesBanner(store);
```
Comment (eselkin) [inline src/applications/static-pages/static-pages-entry.js:204]: Does this need its widget types values or not? The function doesn't seem to need it.

**Review by eselkin — DISMISSED** [review]
LGTM
1 question about common practice/pattern

File: src/applications/static-pages/static-pages-entry.js (hunk 2 above)
Comment (SnowboardTechie) [reply src/applications/static-pages/static-pages-entry.js:204]: Ahhh ya, I built this in a way that does not need it with the createSituationsUpdatesBanner grabbing the widgetType directly. Given this goes against the standard pattern here I'm going to make a quick adjustment to make this lineup better with existing patterns.
Thank you for the question!

**Pull request closed without merging**